docs/source/entities-advanced.mdx

```diff
@@ -0,0 +1,535 @@
+
+In the above example, if a query requests a product's `shippingEstimate`, the gateway does the following, in order:
+
+1. It queries the Products subgraph for the product's `size` and `weight`.
+2. It queries the Shipping subgraph for the product's `shippingEstimate`. The `size` and `weight` are included in the `Product` object passed to the resolver for `shippingEstimate`:
+
+```js{4}
+{
+  Product: {
+    shippingEstimate(product) {
+      return computeShippingEstimate(product.id, product.size, product.weight);
+    }
+  }
+}
+```
+
+### Using `@requires` with object subfields
+
+If a computed field `@requires` a field that returns an object type, you also specify which _subfields_ of that object are required. You list those subfields with the following syntax:
+
+```graphql{4}
+# Shipping subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  dimensions: ProductDimensions @external
+  shippingEstimate: String @requires(fields: "dimensions { size weight }")
+}
+```
+
+In this modification of the previous example, `size` and `weight` are now subfields of a `ProductDimensions` object. Note that the `ProductDimensions` type must be defined in both the Products _and_ Shipping subgraphs for this to be valid.
+
+## Resolving another subgraph's field
+
+By default, exactly one subgraph is responsible for resolving each field in your supergraph schema (with some exceptions, like `@key` fields). But sometimes, _multiple_ subgraphs are able to resolve a particular entity field, because all of those subgraphs have access to a particular data store. For example, an Inventory subgraph and a Products subgraph might both have access to the database that stores all product-related data.
+
+You can enable multiple subgraphs to resolve a particular entity field. **This is a completely optional optimization.** When the gateway plans a query's execution, it looks at which fields are available from each subgraph. It can then attempt to optimize performance by executing the query across the fewest subgraphs needed to access all required fields.
+
+You achieve this with one of the following directives:
+
+* [`@shareable`](#using-shareable)
+* [`@provides`](#using-provides)
+
+Which directive you use depends on the following logic:
+
+```mermaid
+graph TB;
+  resolve("Can my subgraph always resolve this field,<br/>or only from certain query paths?");
+  resolve--Always-->shareable("@shareable");
+  resolve--"Only certain query paths"-->provides("@provides");
+```
+
+If you aren't sure whether your subgraph can always resolve a field, see [Using `@provides`](#using-provides) for an example of a subgraph that _can't_.
+
+> ⚠️ **Important:** If multiple subgraphs can resolve a field, **make sure each subgraph's resolver for that field behaves identically.** Otherwise, queries might return inconsistent results depending on which subgraph resolves the field.
+
+### Using `@shareable`
```

Comment:
I'm not sure this is right place for this section (and this might need a bit of rewording/reworking) because `@shareable` is not specific to entities. In fact, most user will probably use `@shareable` first and mostly to have "shared" value types. So I don't think shareable should be in the "advanced entities" section.

Reply:
Totally, this will not end up the canonical location for learning about `@shareable` (it will be covered in the "value types" article), this is here to accompany `@provides`. We'll have the value type edits up soon and will let you know!